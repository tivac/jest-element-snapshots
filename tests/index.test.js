"use strict";

describe(".toMatchDOMSnapshot()", () => {
    it("should accept a string", async () => {
        await page.goto(require.resolve("./fixtures/one.html"));
        
        await expect(".one").toMatchDOMSnapshot();
    });
    
    it("should accept an ElementHandler", async () => {
        await page.goto(require.resolve("./fixtures/one.html"));

        const el = await page.$(".one");

        await expect(el).toMatchDOMSnapshot();
    });
    
    it("should support named snapshots", async () => {
        await page.goto(require.resolve("./fixtures/one.html"));

        await expect(".one").toMatchDOMSnapshot("named");
    });
});
