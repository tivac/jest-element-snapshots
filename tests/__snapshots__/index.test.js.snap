// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toMatchDOMSnapshot() should accept a string 1`] = `
<div class="one">
  ONE
</div>
`;

exports[`.toMatchDOMSnapshot() should accept an ElementHandle 1`] = `
<div class="one">
  ONE
</div>
`;

exports[`.toMatchDOMSnapshot() should give useful errors when a selector isn't found 1`] = `"Unable to locate element matching selector \\".two\\""`;

exports[`.toMatchDOMSnapshot() should support named snapshots: named 1`] = `
<div class="one">
  ONE
</div>
`;
